assets/src/editor/events/dates-and-times/editor-date/edit-form/date-entity-form-submit-handler.js

```diff
@@ -41,4 +40,5 @@ export const dateEntityFormSubmitHandler = (
 	dateEntity.order = parseInt( getValue( 'order', 0 ), 10 );
 	dateEntity.parent = parseInt( getValue( 'parent', 0 ), 10 );
 	dateEntity.deleted = !! getValue( 'deleted', false );
+	return dateEntity;
```

Comment:
Why return the `dateEntity` here?  Entity instances are mutable so you get back the same instance?  Returning gives the illusion that there's a different value being returned when that's not the case.